src/query/block/meta.go

```diff
@@ -73,32 +75,70 @@ func NewResultMetadata() ResultMetadata {
 	}
 }
 
-// CombineMetadata combines two result metadatas.
-func (m ResultMetadata) CombineMetadata(other ResultMetadata) ResultMetadata {
-	var combinedWarnings Warnings
-	if len(m.Warnings) == 0 {
-		if len(other.Warnings) != 0 {
-			combinedWarnings = other.Warnings
+func combineResolutions(a, b []int64) []int64 {
+	if len(a) == 0 {
+		if len(b) != 0 {
+			return b
 		}
 	} else {
-		if len(other.Warnings) == 0 {
-			combinedWarnings = m.Warnings
+		if len(b) == 0 {
+			return a
 		} else {
-			combinedWarnings = make(Warnings, 0, len(m.Warnings)+len(other.Warnings))
-			combinedWarnings = append(combinedWarnings, m.Warnings...)
-			combinedWarnings = combinedWarnings.addWarnings(other.Warnings...)
+			combined := make([]int64, len(a)+len(b))
+			idx := copy(combined, a)
+			copy(combined[idx:], b)
+			return combined
```

Comment:
Could make this similar to `combineWarnings(...)` to make it more likely for others to use `append(...)` in their own code (if they see it everywhere in the code base, more likely to use it I believe?).

Shrug.

Think it's also compiled down to a memcpy basically anyhow (i.e. `result := append(make([]int64, 0, len(a)+len(b)), a...); result = append(result, b...)` I think should boil down to same assembly output(-ish, i.e. same performance/complexity).

See "BenchmarkAppendAllocInline" on these benchmarks here:
https://gist.github.com/xogeny/b819af6a0cf8ba1caaef#gistcomment-2822939

Reply:
Oh wow nice, will definitely do it this way instead. I guess intuitively it seems that using append would be slower than a straight up copy but nice to see that it's not and we can get free safety checks from just appending (although didn't go do some silly bounds checking on appends?)